### PR TITLE
feat: app settings crud

### PR DESCRIPTION
**Pull request by peterdeme**
- Add app settings crud
- The other output format are coming in another pull request

### CODE REVIEW

File: pkg/cmd/app/app.go [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+
+				indented := bytes.Buffer{}
+				err = json.Indent(&indented, unindented, "", "  ")
+				if err != nil {
+					return err
+				}
+
+				cmd.Println(indented.String())
+			} else {
+				return fmt.Errorf("unknown output format: %s", format)
```
Comment (yaziine) [inline pkg/cmd/app/app.go:48]: For better readability I would suggest to put this before and leave happy path out of `if` statement:
```go
if format != "json" {
	return fmt.Errorf("unknown output format: %s", format)
}

unindented, err := json.Marshal(r.App)
if err != nil {
	return err
}

indented := bytes.Buffer{}
err = json.Indent(&indented, unindented, "", "  ")
if err != nil {
	return err
}

cmd.Println(indented.String())

return nil
```

Feel free to ignore me :)

Comment (yaziine) [reply pkg/cmd/app/app.go:48]: Well this will probably be non relevant in coming PRs when more option will be added.

File: pkg/cmd/app/app.go [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+)
+
+func NewCmds() []*cobra.Command {
+	return []*cobra.Command{getCmd(), updateCmd()}
+}
+
+func getCmd() *cobra.Command {
+	cmd := &cobra.Command{
+		Use: "get-app --output-format [json]",
+		RunE: func(cmd *cobra.Command, args []string) error {
```
Comment (yaziine) [inline pkg/cmd/app/app.go:21]: Missing description (short)

File: pkg/cmd/app/app.go [hunk 3]
```diff
@@ -0,0 +1,96 @@
...
+	fl := cmd.Flags()
+	fl.StringP("output-format", "o", "json", "Output format. Can be json or [see-in-next-pull-request]")
+
+	return cmd
+}
+
+func updateCmd() *cobra.Command {
+	cmd := &cobra.Command{
+		Use: "update-app --properties [raw-json-update-propertes]",
+		Example: heredoc.Doc(`
```
Comment (yaziine) [inline pkg/cmd/app/app.go:64]: same here

File: pkg/cmd/app/app.go [hunk 4]
```diff
@@ -0,0 +1,96 @@
...
+
+	fl := cmd.Flags()
+	fl.StringP("output-format", "o", "json", "Output format. Can be json or [see-in-next-pull-request]")
+
+	return cmd
+}
+
+func updateCmd() *cobra.Command {
+	cmd := &cobra.Command{
+		Use: "update-app --properties [raw-json-update-propertes]",
```
Comment (yaziine) [inline pkg/cmd/app/app.go:63]:
```suggestion
		Use: "update-app --properties [raw-json-update-properties]",
```

File: pkg/cmd/app/app.go [hunk 5]
```diff
@@ -0,0 +1,96 @@
...
+		RunE: func(cmd *cobra.Command, args []string) error {
+			c, err := config.GetConfig(cmd).GetStreamClient(cmd)
+			if err != nil {
+				return err
+			}
+
+			p, _ := cmd.Flags().GetString("properties")
+
+			s := &stream.AppSettings{}
+			err = json.Unmarshal([]byte(p), s)
```
Comment (yaziine) [inline pkg/cmd/app/app.go:76]: Can't we use bytes flag directly?

File: pkg/cmd/app/app.go (hunk 1 above)
Comment (peterdeme) [reply pkg/cmd/app/app.go:48]: yeah exactly, I kinda planned it for the next PRs. and I'll probably extract these to other places

Comment (yaziine) [reply pkg/cmd/app/app.go:48]: ignore me then

File: pkg/cmd/app/app.go (hunk 5 above)
Comment (peterdeme) [reply pkg/cmd/app/app.go:76]: I don't think any of these matches what we want

image

**Review by yaziine — APPROVED** [review]

**Pull request merged**